qrand/platforms/cirq/circuit.py

```diff
@@ -19,85 +19,117 @@
 ## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 ## See the License for the specific language governing permissions and
 ## limitations under the License.
-
+import cirq
+from cirq.circuits import Circuit as CirqQuantumCircuit
 from ..circuit import QuantumCircuit
-
+from ...helpers import validate_natural_number, validate_type
 
 ###############################################################################
 ## CIRQ CIRCUIT
 ###############################################################################
 class CirqCircuit(QuantumCircuit):
+   
     def __init__(self, num_qubits: int) -> None:
-        self.ERROR_MSG = f"{self.__class__.__name__}"  # TODO
-        raise NotImplementedError(self.ERROR_MSG)
-
+        self.qubits=cirq.LineQubit.range(num_qubits)
+        self._num_qubits=num_qubits
+        self.moment_list:list=[]
+    
+    def _validate_qubit_index(self, qubit_index: int) -> None:
+        if qubit_index >= self.num_qubits:
+            raise ValueError(
+                f"Qubit index out of range {qubit_index} >= {self.num_qubits}."
+            )
+
+    
+    
```

Comment:
This has been added as a `QuantumCircuit` method. You can get rid of its declaration here. Keep using it though!